pkg/build/api/types.go

```diff
@@ -507,8 +508,23 @@ type SourceBuildStrategy struct {
 	// Incremental flag forces the Source build to do incremental builds if true.
 	Incremental bool
 
-	// ForcePull describes if the builder should pull the images from registry prior to building.
+	// ForcePull describes if the builder should pull the images from registry
+	// prior to building.
 	ForcePull bool
+
+	// RuntimeImage is an optional image that is used to run an application
+	// without build-time dependencies installed.
+	RuntimeImage *kapi.ObjectReference
+
+	// RuntimeArtifacts holds the mapping between the artifacts paths in the
+	// builder image and the paths in the runtime image where they should be
+	// copied.
+	RuntimeArtifacts []ImageSourcePath
+
+	// RuntimePullSecret is the name of a Secret that would be used for setting up
+	// the authentication for pulling the Docker runtime images from private
+	// Docker registries.
+	RuntimePullSecret *kapi.LocalObjectReference
```

Comment:
yeah that would be another challenge w/ using a single field.

i'm definitely unsure about this since it's relatively unlikely people actually use two different secrets for this..but i could see someone saying "we want to use the openshift EAP image as the runtime image, but we have our own custom maven builder image that's on a separate registry" at which point you are in this scenario.